README.rst

```diff
@@ -385,7 +415,12 @@ call in your custom code (eg: a custom check class), you can do so as follows:
                     "MEAN(buffered) AS buffered FROM {key} WHERE time >= '{time}' AND "
                     "content_type = '{content_type}' AND object_id = '{object_id}' "
                     "GROUP BY time(1d)"
-                )
+                ),
+                'elasticsearch': _make_query({
```

Comment:
why don't you change the code so that calling this functon from the configuration is not necessary?
You can loop over the data structure and call it when it's initialized, this way we make things easy for users and we avoid them come to complain to us in the support channels :joy:

Reply:
Yes that can be done and I have done the same for built-in charts. `_make_query` is just a utility function which will update the aggregation for a `default_chart_query` defined in **openwisp_monitoring.db.backends.elasticsearch.queries**. So queries returned via `_make_query` will always retain the structure of `default_chart_query`.

I wanted to leave the option of directly using a dsl query with `timeseries_db.query`, exactly like how we can query InfluxDB directly using the same function.

A full **dsl** query will look like this,
```python
{'query': {'nested': {'path': 'tags',
   'query': {'bool': {'must': [{'match': {'tags.object_id': {'query': '9a39a5ae-146b-4a50-b113-f9381b8c1721'}}},
      {'match': {'tags.content_type': {'query': 'config.device'}}}]}}}},
 '_source': False,
 'size': 0,
 'aggs': {'GroupByTime': {'nested': {'path': 'points',
    'aggs': {'set_range': {'filter': {'range': {'points.time': {'from': 'now-1d/d',
         'to': 'now/d'}}},
      'aggs': {'time': {'date_histogram': {'field': 'points.time',
         'fixed_interval': '10m',
         'format': 'date_time_no_millis',
         'order': {'_key': 'desc'},
         'time_zone': 'Asia/Kolkata'},
        'aggs': {'nest': {'nested': {'path': 'points.fields',
           'aggs': {'CPU_load': {'avg': {'field': 'points.fields.cpu_usage'}}}}}}}}}}}}}}
```

So, if we make `_make_query` as a compulsion, we might be cutting down a user's freedom to query via **DSL**. Personally, I would like to give user this freedom (this would enable him to just put a query like above in chart configuration and it will work) :smile:.